packages/react-components/react-progress/src/components/Progress/useProgressStyles.ts

```diff
@@ -1,33 +1,201 @@
+    color: tokens.colorNeutralForeground2,
+  },
+});
+
+/**
+ * Styles for the progress bar
+ */
+const useBarStyles = makeStyles({
+  base: {
+    gridColumnStart: '1',
+    gridRowStart: '2',
+    backgroundColor: tokens.colorCompoundBrandBackground,
+
+    '@media screen and (forced-colors: active)': {
+      backgroundColor: 'Highlight',
+    },
+  },
+  medium: {
+    height: barThicknessValues.medium,
+  },
+  large: {
+    height: barThicknessValues.large,
+  },
+  determinate: {
+    width: `var(${progressCssVars.percentageCssVar})`,
+  },
+  nonZeroDeterminate: {
+    width: `var(${progressCssVars.percentageCssVar})`,
+    transitionProperty: 'width',
+    transitionDuration: '0.3s',
+    transitionTimingFunction: 'ease',
+  },
+  indeterminate: {
+    maxWidth: '33%',
+  },
+  ltr: {
+    position: 'relative',
+    backgroundImage: `linear-gradient(
+      to right,
+      ${tokens.colorNeutralBackground6} 0%,
+      ${tokens.colorCompoundBrandBackground} 50%,
+      ${tokens.colorNeutralBackground6} 100%
+    )`,
+    animationName: IndeterminateProgress,
+    animationDuration: '3s',
+    animationIterationCount: 'infinite',
+  },
+
+  rtl: {
+    animationDirection: 'reverse',
   },
+});
+
+const useTrackStyles = makeStyles({
+  base: {
+    gridRowStart: '2',
+    gridColumnStart: '1',
+    backgroundColor: tokens.colorNeutralBackground6,
 
-  // TODO add additional classes for different states and/or slots
+    '@media screen and (forced-colors: active)': {
+      ...shorthands.borderBottom('1px', 'solid', 'CanvasText'),
+    },
+  },
+  medium: {
+    height: barThicknessValues.medium,
+  },
+  large: {
+    height: barThicknessValues.large,
+  },
 });
 
 /**
  * Apply styling to the Progress slots based on the state
  */
 export const useProgressStyles_unstable = (state: ProgressState): ProgressState => {
-  const styles = useStyles();
-  state.root.className = mergeClasses(progressClassNames.root, styles.root, state.root.className);
+  const { indeterminate = false, thickness = 'medium', percentComplete = 0 } = state;
+  const rootStyles = useRootStyles();
+  const barStyles = useBarStyles();
+  const trackStyles = useTrackStyles();
+  const labelStyles = useLabelStyles();
+  const descriptionStyles = useDescriptionStyles();
+  const { dir } = useFluent();
+
+  const valuePercent = !indeterminate ? Math.min(100, Math.max(0, percentComplete! * 100)) : undefined;
+  const progressBarStyles = {
+    [`${progressCssVars.percentageCssVar}`]: !indeterminate ? valuePercent + '%' : undefined,
+  };
+
+  state.root.className = mergeClasses(progressClassNames.root, rootStyles.root, state.root.className);
+
+  // Determinate indicator when determinate is defined
+  if (state.bar) {
+    state.bar.className = mergeClasses(
+      progressClassNames.bar,
+      barStyles.base,
+      indeterminate && barStyles.indeterminate,
+      indeterminate && barStyles[dir],
+      barStyles[thickness],
+      !indeterminate && barStyles.determinate,
```

Comment:
Do I correctly understand that `Progress` is `indeterminate` by default? In this case we can move `barStyles.indeterminate` to `barStyles.base`:

```diff
-  indeterminate: {
-    maxWidth: '33%',
-  },
```
```diff
    gridRowStart: '2',
    backgroundColor: tokens.colorCompoundBrandBackground,
+   maxWidth: '33%',
```
```diff
      barStyles.base,
-     indeterminate && barStyles.indeterminate,
```

Reply:
@layershifter Wouldn't that require overriding the `maxWidth` property for the determinate state? Determinate doesn't set `maxWidth`.

I generally prefer "additive" styles, where the `base` style contains props that are common to all states, and then there is a style for each state (including one for the default state, if needed) that defines the props that are unique to that state. There's a lot of wiggle room in terms of exactly how to implement that, but in general a state's styles shouldn't have to remove/override a style prop that isn't applicable to it.

Reply:
Also, `determinate` is now the default state, so `maxWidth: 33%` shouldn't be set as a base style

Reply:
@behowell Oops, indeed 🫢

![image](https://user-images.githubusercontent.com/14183168/193325056-0de62365-60e9-4f18-8960-c1f0d897d97a.png)